contrib/babelfishpg_tsql/src/pltsql_coerce.c

```diff
@@ -638,9 +640,26 @@ init_tsql_coerce_hash_tab(PG_FUNCTION_ARGS)
 
 						if (!OidIsValid(entry->castfunc))
 						{
+							/*
+							 * varbinary to binary implicit type cast without function should be allowed
+							 * during upgrade since the cast function might not exists in older version
+							 */ 
+							if (babelfish_dump_restore && ((*common_utility_plugin_ptr->is_tsql_varbinary_datatype) (castsource) 
+								&& (*common_utility_plugin_ptr->is_tsql_binary_datatype) (casttarget)))
+							{
+								entry->castfunc = 0;
+								entry->castcontext = COERCION_CODE_IMPLICIT;
+								entry->castmethod = COERCION_METHOD_BINARY;
+								value = hash_search(ht_tsql_cast_info, key, HASH_ENTER, NULL);
+								*(tsql_cast_info_entry_t *) value = *entry;
```

Comment:
how this entry will be replaced by actual function once minor upgrade happens?

Reply:
after the upgrade script has been run, we reload this hash map and then we will have the new function available.